modules/core/src/webgl/classes/buffer.js

```diff
@@ -3,6 +3,7 @@ import Resource from './resource';
 import Accessor from './accessor';
 import {assertWebGL2Context, getGLTypeFromTypedArray, getTypedArrayFromGLType} from '../utils';
 import {log, assert, checkProps} from '../../utils';
+import statsManager from '../../core/stats-manager';
```

Comment:
Note: Buffers can be `reallocated`, also check `setData`/`_setData` to follow how they reinitialize buffers. Might want to keep things updated.

Reply:
So I believe this is covered by the fact it always subtracts the current `byteLength` from the counts before adding to the count after allocation. Is there a path that I'm missing?

Reply:
OK. Worth double checking `Buffer._setByteLength` I think it can change the size of the buffer.

Reply:
Ah good catch! Wasn't covering the `reallocate` method. Moving tracking into `_setData` and `_setByteLength`